src/averageAges.js

```diff
@@ -36,8 +46,14 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
-function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+function calculateWomenAverageAge(people, withChildren = false) {
+  const female = people.filter(person => person.sex === 'f');
+  const hasChildren = people.filter((person) =>
+    people.find((child) => person.name === child.mother)
+  );
+  const result = withChildren ? hasChildren : female;
```

Comment:
U really need to use the filter two times🤔

Now complexity is O(2n), go try to implement this with O(n)